service/templates/notifications.html

```diff
@@ -86,27 +88,34 @@
 
     <md-card flex="60">
       <md-card-title>
-        <md-card-title-text>
-          Existing Subscriptions
-        </md-card-title-text>
+        <md-card-title-text>Manage Subscriptions</md-card-title-text>
       </md-card-title>
       <md-card-content>
-
         <table class="table table-new">
           <thead>
             <tr>
               <th class="col-lg-3">Selectors</th>
-              <th class="col-lg-3">Trigger</th>
-              <th class="col-lg-3">Action</th>
-              <th class="col-lg-1"> </th>
+              <th class="col-lg-3">Event</th>
+              <th class="col-lg-3">Notify by</th>
+              <th class="col-lg-1"></th>
             </tr>
           </thead>
           <tbody>
             <tr ng-repeat="subscription in subscriptions"
                 class="test-result-row">
               <td class="col-lg-3">
-                <a href="javascript:void(0);" ng-click="selectorClick([[subscription.selectors]])"
-                 ng-bind="selectorText([[subscription.selectors]])"> </a>
+                  <span ng-repeat="s in subscription.selectors">
+                    <p ng-show="s.type === 'object'">when a [[s.data]] with:</p>
+                    <p ng-show="s.type !== 'object' && (s.type === 'id' || s.type.startsWith('in-'))"><b>[[s.type]]</b> of
+                      <span style="width: 50%; display: block; overflow: hidden; text-overflow: ellipsis;"><a href="javascript:void(0);" ng-click="selectorClick([[subscription.selectors]])">[[s.data]]</a></span>
```

Comment:
The reason I didn't want to display IDs is because they are meaningless and really long. If you want to include more contextual info about the subscription, I think you need to display either name or commit message

Reply:
What about for patches that have no description?

Reply:
The ID is only meaningless in the case of version and patch. For all others, we encode some combination of project, task name, and build variant into it. I can add descriptions on top of what's there already, but I'm still in favour of including the ids.